samples/crypto/aes_ccm/prj.conf

```diff
@@ -16,4 +16,5 @@ CONFIG_MBEDTLS_ENABLE_HEAP=y
 CONFIG_MBEDTLS_HEAP_SIZE=8192
 
 CONFIG_PSA_WANT_GENERATE_RANDOM=y
+CONFIG_PSA_WANT_KEY_TYPE_AES=y
 CONFIG_PSA_WANT_ALG_CCM=y
```

Comment:
Our drivers only support the AES key for CCM.

Would it be possible for us to select the AES key from the CCM algorithm?

Or maybe some other mechanism to not make this "choice" that is not really a choice.
That would mean less config work for our users.

![image](https://github.com/nrfconnect/sdk-nrf/assets/609230/3345625b-d5d8-451b-8408-e6298b164ce8)

Reply:
Maybe this is a case for the Kconfig "imply" mechanism?

Reply:
Hi, I did some experiments to see if this would work and it appears to work as intended.

https://github.com/SebastianBoe/fw-nrfconnect-nrf-1/commit/adc80da258298d5b9142759b056be1f48b846f4f

Here we introduce a promptless default value for the AES key type that is processed early in the configuration to override upstream configs. This follows prior-art (openthread).

I agree we should not patch upstream config files with our platform-specific limitations.

"I find it simplest to have a one-to-one mapping between Kconfig symbols to header defines, instead of having implicit symbols in this way."

I agree with this and in the proposed patch we would still have a one-to-one mapping between Kconfig symbols and header defines.

"Also with the PSA_WANT_AES_KEY_SIZE_128/192/256 it becomes kind of strange to not have the key type for AES IMO."

I agree that we should have a Kconfig symbol for the AES key type, in the proposed patch we still have such a symbol.